### PR TITLE
Add docs and small fixits

### PR DESCRIPTION
**Pull request by jackbackes**
Mostly for my own benefit so I can understand the codebase, adds substantial documentation to all publicly exposed methods. This commit also introduces `strum` for consistent enum serialization and the ability to add props to enums. This will allow us to keep "dimensions" (or any other properties) along with the other embedding model features.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,28 +7,30 @@ edition = "2021"
 members = [".", "migration"]
 
 [dependencies]
-rust-bert = {version="0.20.0"}
```
Comment (jackbackes) [inline Cargo.toml:10]: sorted and added "strum" and "strum_macros"

File: src/server.rs [hunk 2]
```diff
@@ -15,63 +15,72 @@ use std::collections::HashMap;
...
+    /// Split HTML text into a specified number of elements.
     Html {
         #[default = 1]
-        #[serde(rename = "elements")]
+        #[strum(serialize = "elements")]
         num_elements: u64,
     },
 }
 
-impl fmt::Display for TextSplitterKind {
```
Comment (jackbackes) [inline src/server.rs:54]: Replaced with strum::Display derive macro

File: src/server_config.rs [hunk 3]
```diff
@@ -4,82 +4,100 @@ use figment::{
...
 const OPENAI_DUMMY_KEY: &str = "xxxxx";
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Enum representing the different kinds of text embedding models available for use.
+/// Each variant is associated with specific dimensions, which represent the size of the embeddings.
+#[derive(Debug, Clone, Serialize, Deserialize, strum_macros::Display, strum_macros::EnumProperty)]
+#[strum(serialize_all = "kebab-case")]
 pub enum EmbeddingModelKind {
-    #[serde(rename(serialize = "all-minilm-l12-v2", deserialize = "all-minilm-l12-v2"))]
+    #[strum(props(dimensions = "384"))]
```
Comment (jackbackes) [inline src/server_config.rs:17]: Not using Props yet but can replace the extra "match" blocks in the embeddings sub-crate

**Comment by diptanu** [comment]
LGTM!

File: src/server_config.rs [hunk 4]
```diff
@@ -4,82 +4,100 @@ use figment::{
...
+    #[strum(serialize = "all-minilm-l12-v2")]
     AllMiniLmL12V2,
 
-    #[serde(rename(serialize = "all-minilm-l6-v2", deserialize = "all-minilm-l6-v2"))]
+    #[strum(props(dimensions = "384"))]
+    #[strum(serialize = "all-minilm-l6-v2")]
     AllMiniLmL6V2,
 
-    #[serde(rename(serialize = "t5-base", deserialize = "t5-base"))]
```
[the remark this answers is not in the document]
Comment (diptanu) [reply src/server_config.rs:20]: They are different models. We don't have the model all-mpnet-base-v2 yet.

**Comment by diptanu** [comment]
@jackbackes How does strum work with serde? I had to add back the serde related macros to be able to parse the server config correctly.

**Pull request merged**